sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -606,20 +611,28 @@ public void setSessionState(String sessionId, byte[] sessionState) {
     @Override
     public void close() {
         asyncClient.close();
+
+        if (messageProcessor.get() != null) {
```

Comment:
The operation should be `cancel()`, not `onComplete()`. onComplete and methods like that should be done through a sink.

```java
var processor = messageProcessor.getAndSet(null);
if (processor != null) {
   processor.cancel()
}
```

Reply:
discussed on teams and decided to not change it.